pulumi/src/coinstack.ts

```diff
@@ -0,0 +1,145 @@
+import { Config, deployApi, JointCoinServiceInput, Outputs, SecretData, ServiceConfig, ServiceInput, Snapper } from '.'
+import { getConfig } from './config'
+import { parse } from 'dotenv'
+import { createService, deployStatefulService } from './statefulService'
+import * as k8s from '@pulumi/kubernetes'
+import { CoinstackType } from './hash'
+
+export const deployCoinstack = async (
+  appName: string,
+  coinstack: string,
+  serviceInput: ServiceInput[],
+  sampleEnv: Buffer,
+  coinstackType: CoinstackType,
+  volumes?: Array<k8s.types.input.core.v1.Volume>
+): Promise<Outputs> => {
+  const { kubeconfig, config, namespace } = await getConfig()
+
+  const assetName = config.network !== 'mainnet' ? `${config.assetName}-${config.network}` : config.assetName
+  const provider = new k8s.Provider('kube-provider', { kubeconfig })
+  const secretData = getSecretData(sampleEnv)
+
+  new k8s.core.v1.Secret(assetName, { metadata: { name: assetName, namespace }, stringData: secretData }, { provider })
+
+  const baseImageName = 'shapeshiftdao/unchained-base:latest'
+  const snapshots = await new Snapper({ assetName, kubeconfig, namespace }).getSnapshots()
+
+  await deployApi({
+    appName,
+    assetName,
+    coinstack,
+    coinstackType,
+    sampleEnv,
+    baseImageName,
+    config,
+    namespace,
+    provider,
+  })
+
+  const coinServices = aggregateCoinServiceInput(config.statefulService?.services || [], serviceInput)
+    .map((cs) => enrichWithPulumiConfig(cs, namespace, coinstack, config))
+    .map((cs) => createService(cs, assetName, snapshots))
+  await deployStatefulService(appName, assetName, provider, namespace, config, coinServices, volumes)
+  return {}
+}
+
+const enrichWithPulumiConfig = (
```

Comment:
We should rethink this. Why not include this as part of the top level service config?

Reply:
along these lines and can include `env` at that point
![image](https://user-images.githubusercontent.com/35275952/235494911-e48770e4-338c-4d4b-8c82-a15a3b5b1ae5.png)

Reply:
But that's the essence of the problem. The config parsing does not happen where input arguments are defined, so we can't reference config variables until already inside `deployCoinstack`

Reply:
That's why I'm suggesting parsing the config one level up and passing it down to `deployCoinstack`. Then we have the `namespace` where needed and can have the env vars set explicitly for each coinstack